src/types.rs

```diff
@@ -344,17 +389,42 @@ impl PlaceOrder {
                 PostOnlyParam::None
             },
             user_order_id: self.user_order_id,
-            oracle_price_offset: self
-                .oracle_price_offset
-                .map(|x| scale_decimal_to_i64(x, PRICE_PRECISION as u32) as i32),
+            oracle_price_offset,
             max_ts: self.max_ts,
             trigger_price: self
                 .trigger_price
                 .map(|v| scale_decimal_to_u64(v, PRICE_PRECISION as u32)),
             trigger_condition: self.trigger_condition.unwrap_or_default(),
+            auction_duration: Some(self.auction_duration.unwrap_or(20)),
+            auction_start_price: self.auction_start_price,
+            auction_end_price: self.auction_end_price,
             ..Default::default()
         }
     }
+
+    pub fn to_signed_order_hex(
+        self,
+        order_params: OrderParams,
+        slot: u64,
+        sub_account_id: u16,
+    ) -> Vec<u8> {
+        let order = SignedMsgOrderParamsMessage {
+            signed_msg_order_params: order_params,
+            slot,
+            uuid: nanoid!(8).as_bytes().try_into().unwrap(),
+            sub_account_id,
+            take_profit_order_params: None, // TODO: add take profit order params
+            stop_loss_order_params: None,   // TODO: add stop loss order params
+        };
+
+        let signed_order_type = SignedOrderType::Authority(order);
+
+        let borsh_encoding = signed_order_type.to_borsh();
+        let borsh_bytes = borsh_encoding.as_slice();
+        let mut hex_bytes = vec![0; borsh_bytes.len() * 2]; // 2 hex bytes per msg byte
+        let _ = faster_hex::hex_encode(borsh_bytes, &mut hex_bytes).expect("hexified");
+        hex_bytes.as_slice().to_owned()
```

Comment:
```suggestion
        hex_bytes
```
or clone is better if that doesn't work